libraries/tickv/src/async_ops.rs

```diff
@@ -193,18 +195,30 @@ impl<'a, C: FlashController<S>, const S: usize> AsyncTicKV<'a, C, S> {
         &self,
         hash: u64,
         value: &'static mut [u8],
-    ) -> Result<SuccessCode, (Option<&'static mut [u8]>, ErrorCode)> {
-        match self.tickv.append_key(hash, value) {
-            Ok(code) => Ok(code),
+        length: usize,
+    ) -> Result<SuccessCode, (&'static mut [u8], ErrorCode)> {
+        match self.tickv.append_key(hash, &value[0..length]) {
+            Ok(_code) => {
+                // Ok is a problem, since that means no asynchronous operations
+                // were called, which means our client will never get a
+                // callback. We need to error.
+                Err((value, ErrorCode::WriteFail))
+            }
```

Comment:
This seems better than how things worked before, though I do think we might eventually want to support an API that can sit on top of maybe-async operations, which would rely on this library actually scheduling a callback in the case that the underlying `append_key()` happened synchronously. I don't want to block this on that though, as I think getting these changes in so things are in a workable state is important.

Reply:
Hmm, that makes sense. That does get fairly complicated, however. Since the caller has to implement both the lower layer (flash) and upper layer (in our case tickv capsule), it would be strange for the caller to both implement a synchronous flash but then try to use the async API.

Reply:
Makes sense. Lets stick with this for now and can change it if/when a use case comes along